### PR TITLE
adding handler name to user service metrics ccontext tags

### PR DESCRIPTION
**Pull request by gsps1**
we have metrics of requests to /response from user services, 
this PR is to add the user service handler detail to the metrics context. 
https://issues.cask.co/browse/CDAP-2181

NOTE: we don't add method information to the context right now, but all user services metrics currently are emitted at handler level.

### CODE REVIEW

File: cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java [hunk 1]
```diff
@@ -432,6 +432,14 @@ public void testAppWithServices() throws Exception {
     Assert.assertEquals(2, serviceMetrics.getProcessed());
     Assert.assertEquals(1, serviceMetrics.getException());
 
+    RuntimeMetrics handlerMetrics = RuntimeStats.getServiceHandlerMetrics(AppWithServices.APP_NAME,
+                                                                          AppWithServices.SERVICE_NAME,
+                                                                          AppWithServices.SERVICE_NAME);
```
Comment (abaranau) [inline cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java]: AppWithServices.SERVICE_NAME twice? typo? how does test work then? :)

**Comment by abaranau** [comment]
just one Q, after fix - LGTM

File: cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java (hunk 1 above)
Comment (gsps1) [inline cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java]: in the test app, they use the same name for service and handler, that's why.

**Comment by gsps1** [comment]
http://builds.cask.co/browse/CDAP-DUT1571-4/test - GREEN

File: cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java (hunk 1 above)
Comment (abaranau) [inline cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java]: add a comment then :) Since someone asked - it is confusing :)

**Comment by abaranau** [comment]
please add comment and merge!

**Pull request merged**